pipert/core/message.py

```diff
@@ -0,0 +1,107 @@
+from abc import ABC, abstractmethod
+from io import BytesIO
+
+import numpy as np
+import time
+import pickle
+import cv2
+
+from PIL import Image
+
+
+class Payload(ABC):
+
+    def __init__(self, data):
+        self.data = data
+        pass
+
+    @abstractmethod
+    def decode(self):
+        pass
+
+    @abstractmethod
+    def encode(self):
+        pass
+
+
+class FramePayload(Payload):
+
+    def __init__(self, data):
+        super().__init__(data)
+
+    def decode(self):
+        self.data = np.array(Image.open(BytesIO(self.data)))
+
+    def encode(self):
+        _, buf = cv2.imencode(".jpg", self.data)
+        self.data = buf.tobytes()
+
+
+class PredictionPayload(Payload):
+    def __init__(self, data):
+        super().__init__(data)
+
+    def decode(self):
+        pass
+
+    def encode(self):
+        pass
+
+
+class Message:
+    next_msg_id = 0
+
+    def __init__(self, data, stream_address):
```

Comment:
It's not necessarily a video stream. Maybe change the name to `data_source` or something like that. Also, could it be optional? @sionovd

Reply:
Do you mean to say that it could also originate in a video file/webcam stream? Or are you talking about something else? 
And why should it be optional if every message is linked to some video source? @ItamarWilf

Reply:
I'd appreciate some clarification here @ItamarWilf

Reply:
Yes, what I meant is the pipeline isn't restricted to video applications, and the data could originate from many sources, for example, streaming metadata from an external source. This small change is only for user readability :)